tululbot/utils.py

```diff
@@ -20,9 +40,35 @@ def retrieve_random(self):
     def format_quote(self, q):
         return '{q[quote]} - {q[author]}, {q[author_bio]}'.format(q=q)
 
-    def refresh_cache(self):
+    def refresh_cache_if_applicable(self):
+        if (not self.refresh_url_if_applicable()):
+            return False
```

Comment:
why this method has to return boolean? I look at the usage and this return value is ignored by the caller anyway.

as a side note, my preference is to separate pure functions (no side-effect) and impure functions. pure functions should return a meaningful value to the caller while impure ones shouldn't. this is like procedures in Pascal. I find this separation makes the functions or methods easier to reason about.